### PR TITLE
Select a build to compare to

### PR DESCRIPTION
**Pull request by cdcarter**
when viewing a build's [test limits usage](https://mrbelvedereci.herokuapp.com/tests/6187/ci_feature) this PR adds a Compare button, that displays a list of recent builds to compare it to.

test compare

adds a new page & view to list compare options, a helper to lookup BuildFlow based on the tests/ url patterns, and uses django_filters to filter the queryset *and* generate a form for the client.

### CODE REVIEW

File: mrbelvedereci/testresults/urls.py [hunk 1]
```diff
@@ -5,6 +5,11 @@
 
 urlpatterns = [
     url(
+        r'^(?P<build_id>\d+)/(?P<flow>.*)/compare_to',
```
Comment (cpolcyn) [inline mrbelvedereci/testresults/urls.py]: I'd like this changed to `compare-to` but is just my preference.

File: mrbelvedereci/testresults/utils.py [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+from mrbelvedereci.build.models import Build
+from mrbelvedereci.build.models import BuildFlow
+
+
+def find_buildflow(request, build_id, flow):
+    """ given a build_id and flow name, find a single BuildFlow (ala tests/ urls patterns). """
+    build = get_object_or_404(Build, id=build_id)
+
+    if not build.plan.public and not request.user.is_staff:
+        raise Http404()
```
Comment (cpolcyn) [inline mrbelvedereci/testresults/utils.py]: Should be 403 Forbidden

File: requirements/base.txt [hunk 3]
```diff
@@ -11,6 +11,7 @@ whitenoise==3.2.2
 # Forms
 django-braces==1.10.0
 django-crispy-forms==1.6.1
+django_filter==1.0.4
```
Comment (cpolcyn) [inline requirements/base.txt]: Why is this needed? Can't this be done simply using URL query params passed in the request and use Django object query?

File: mrbelvedereci/testresults/utils.py [hunk 4]
```diff
@@ -0,0 +1,17 @@
+from django.shortcuts import get_object_or_404
+from mrbelvedereci.build.models import Build
+from mrbelvedereci.build.models import BuildFlow
+
+
+def find_buildflow(request, build_id, flow):
+    """ given a build_id and flow name, find a single BuildFlow (ala tests/ urls patterns). """
+    build = get_object_or_404(Build, id=build_id)
+
+    if not build.plan.public and not request.user.is_staff:
```
Comment (cpolcyn) [inline mrbelvedereci/testresults/utils.py]: `is_staff` limits to only admins – maybe you want `is_authenticated`?

Comment (cdcarter) [inline mrbelvedereci/testresults/utils.py]: <redacted_adminsite_url>/users/user/?is_staff__exact=1 - `is_staff` is internal users, `is_superuser` is admins. 

an auth helper for builds in general would be nice, and I know we have a ticket for using github access...

File: requirements/base.txt (hunk 3 above)
Comment (cdcarter) [inline requirements/base.txt]: the library (and filter subclass) is also generating the form on the page (search fields for build name and plan name).

it's not strictly necessary, but it meant I didn't have to build a form. would it be worth it for me to try the other method and compare?

File: mrbelvedereci/testresults/utils.py [hunk 5]
```diff
@@ -8,7 +9,7 @@ def find_buildflow(request, build_id, flow):
     build = get_object_or_404(Build, id=build_id)
 
     if not build.plan.public and not request.user.is_staff:
-        raise Http404()
+        raise Http403()
```
Comment (cpolcyn) [inline mrbelvedereci/testresults/utils.py]: `Http403` doesn't exist, should be `HttpResponseForbidden`. Sorry if my previous comment was unclear!

Comment (cdcarter) [inline mrbelvedereci/testresults/utils.py]: No, that's my bad. my server is complaining and everything.

File: requirements/base.txt (hunk 3 above)
Comment (cpolcyn) [inline requirements/base.txt]: The in-page form is cool and seems worth it for that. Thanks for clarifying!
